client/container_exec.go

```diff
@@ -14,6 +14,9 @@ func (cli *Client) ContainerExecCreate(ctx context.Context, container string, co
 	if err := cli.NewVersionError("1.25", "env"); len(config.Env) != 0 && err != nil {
 		return response, err
 	}
+	if err := cli.NewVersionError("1.42", "console size"); config.ConsoleSize != nil && err != nil {
+		return response, err
+	}
```

Comment:
Perhaps for this one we shouldn't produce an error (we are a bit inconsistent on this though!); my train of thought here is that; unlike `env` above, the console size is not an explicit option set by the user; this option is a bit similar to how we moved the "auto remove"  (`--rm`) from the client to the daemon; https://github.com/moby/moby/blob/b6dab55339382e2b6df0678a511a75c146daa45b/client/container_create.go#L32-L35

Having this produce an error means that we also need to check the API version in `docker/cli`. Perhaps instead we could just set it unconditionally in the CLI (and have the client discard the value), which could be slightly easier to use.

Happy to hear your (and other's) thoughts on that as well though!